processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -811,7 +811,11 @@ public Optional<Sequence<FrameSignaturePair>> resultsAsFrames(
       boolean useNestedForUnknownTypes
   )
   {
-    RowSignature rowSignature = resultArraySignature(query);
+    RowSignature rowSignature = query.getResultRowSignature(
+        query.context().isFinalize(true)
+        ? RowSignature.Finalization.YES
+        : RowSignature.Finalization.NO
+    );
```

Comment:
I believe this logic supposed to be inside `query.getResultRowSignature`; `query` already knows `context()` ; why should we tell it from the outside the value of `Finalization`?

Reply:
It should work, but I am scared to make that change given that it will affect everything from native and SQL queries. Lemme try making the change and see if there are any failing tests.

Reply:
I realised that it shouldn't work. For example - look at [GroupByPreShuffleFrameProcessor](https://github.com/apache/druid/blob/master/extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByPreShuffleFrameProcessor.java#L100) and [GroupByPostShuffleFrameProcessor](https://github.com/apache/druid/blob/master/extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByPostShuffleFrameProcessor.java#L107). The same query requires different finalization modes, since one partially aggregates and we need to intermediate type while the other completely aggregates and finalizes. This information isn't fully captured by the query and needs someone from the outside to tell which finalization mode to use. Therefore we can't trustily determine based on the query context.